src/utils/affine.ts

```diff
@@ -21,4 +21,13 @@ export namespace Affine {
     export function createVector(vector: vec3): vec4 {
         return vec4.fromValues(vector[0], vector[1], vector[2], 0);
     }
+
+    /**
+     * Convert a vector from Affine space to 3D space.
+     * @param {vec4} vector: The vector to truncate.
+     * @return {vec3}: The vector without the last element.
+     */
+    export function truncate(vector: vec4): vec3 {
```

Comment:
Can you rename this to something like `affineTo3D`? I think calling this `truncate` is a bit confusing, since the logical operation is a conversion from one set to another, and not chopping off the last component of the vector..

Reply:
Renamed to `to3D` since it should be called with the `Affine` prefix anyway `Affine.to3D(vec)`.